### ISSUE TITLE
List packages for KDE

### ISSUE THREAD

**Issue by varlesh** [Antergos/antergos-packages#95]
Why on KDE desktop installed this packages as default?
lightdm - but for KDE recommends SDDM. Why not use SDDM?
pamac - Pamac GTK. But KDE use almost Qt-apps. Why not Octopi?
kipi-plugins - this really not needed package as default!

**Comment by karasu** [comment]
Hi,

We use lightdm in all DE's, so it has the same look and feel.

pamac - I agree. We use pamac as default, but I think Openbox uses Octopi (I can't remember). We could change this. Let me talk with the other devs

kipi-plugins : I did not package KDE, so I'm not sure why is there or if it's really needed. Again, I will talk with the others and get back to you

**Comment by varlesh** [comment]
Thx for the answer :)
Ok, Lightdm everywhere. But, kipi-plugins really not needed. Also octopi - very good and stable GUI for pacman.
Maybe kipi-plugins install as depend?

**Comment by varlesh** [comment]
Developers please tell my why need this packages as default on [antergos-kde-setup](https://github.com/Antergos/antergos-packages/blob/master/antergos-kde-setup/PKGBUILD)?
kfaenza-icon-theme - Last Updated:  [Nov 5 2011](http://kde-look.org/content/show.php?content=143890). Five years ago!!! Numix and Breeze enough.
pamac - why not Octopi?
kipi-plugins - really not needed!

**Comment by Paviluf** [comment]
I have some though too about the KDE packages

- Make some softs optionnal https://github.com/Antergos/antergos-kde-setup/issues/9
- Use sddm instead of lightdm https://github.com/Antergos/antergos-kde-setup/issues/10

Thanks !

**Comment by Paviluf** [comment]
kipi-plugins may be [useful](https://en.wikipedia.org/wiki/KDE_Image_Plugin_Interface) but clearly not kfaenza-icon-theme

**Comment by karasu** [comment]
@lots0logs this is why I don't like KDE installation not having all necessary packages specified in packages.xml (cnchi) and make it all depend on this package.

We can talk about this in private.

**Comment by lots0logs** [comment]
Sure we can discuss it. To address what was recently added to this issue though, I don't think we want to provide loads of options. Its not good for the user experience (using Cnchi). Anyway, like I said we can discuss it further for sure. Thanks for your input @Paviluf ! :smiley:

**Comment by slacka** [comment]
Agreed, install options should be kept to a minimum. The improvements should be based around choosing better defaults and packages to provide the best out of box experience. The beauty of Linux is the user can always customize after the install. Too many install options confuse the user and make qa/testing that much harder.

**Comment by Paviluf** [comment]
Any progress ? Thanks !

**Comment by karasu** [comment]
@lots0logs you're still against removing all dependencies from antergos-kde-setup? (I'm just asking).

**Comment by lots0logs** [comment]
@karasu Affirmative (sorry!). And I take it you are still against implementing all the desktops in the way that KDE is currently implemented?

**Comment by karasu** [comment]
@lots0logs
>And I take it you are still against implementing all the desktops in the way that KDE is currently implemented?

Yes (sorry!) 😬 

I see your point. But it's confusing for the users when they try to remove a package that depends on antergos-whatever-setup.

>Doing so will not remove the configuration that has already been done. It will simply allow one to alter the selection of packages to their liking

But they tend to think that something will break if they remove the antergos setup package.

On the other hand, if we want to update our setup, many users will not have the package installed, so our changes will only affect new installations.

In my oppinion the antergos setup packages should only apply our setup and leave all dependencies optional. I understand that this defeats the purpose of using them when installing, but this way we give the users the option to modify packages.xml themselves.

Each option has its pros and cons.

**Comment by lots0logs** [comment]
@karasu I think you still don't understand my point. 

>In my oppinion the antergos setup packages should only apply our setup and leave all dependencies optional.

This is exactly what they do. All you have to do is remove the setup package if you want to remove something that it depends on. Doing so is not going to remove all the other packages. You lose nothing by removing it. As I said, the setup package is intended for installing the default config for the desktop. It is not intended for long-term maintenance of said configuration. 

In fact, we could just as easily remove the setup package immediately after installing it. I dont think doing that is necessary personally, because removing one package when you want to remove something it depends on is a trivial task. One which any user should be perfectly capable of doing (especially if they know enough to have come to the conclusion that want to remove one of the default packages anyway). Its harmless.

That being said, if you still disagree with my point (honestly I dont know how else to explain it, I'm having a hard time understanding how one could disagree with it) HOWEVER, I don't need to understand in order to accept and respect it so its your call :smiley:

**Comment by lots0logs** [comment]
>but this way we give the users the option to modify packages.xml themselves.

Honestly that is such an edge use-case that I dont think we should be worried about that. Besides, anyone who wants to do that is going to know how to do it regardless of whether or not each and every package is listed in the packages.xml file. Its not difficult (obviously).

**Comment by Yoshi2889** [comment]
I think a balance can be found between the usage of antergos-{de}-setup and keeping dependencies to a minimum. Maybe provide a way in cnchi to install all optional dependencies of a package, and then create an antergos-{de}-addons package? That way we could include the whole KDE experience, but allow the user to uninstall components cleanly.

Therefore, I propose the antergos-{de}-setup package be split:

- antergos-{de}-base: Critical components of each DE (e.g. plasma-desktop for Plasma)
- antergos-{de}-config: Antergos specific configuration filesof each component; this should not overwrite existing config files where possible.
- antergos-{de}-addons: Like described above, any non-critical addons or applications that were thought of to be good defaults.

This potentially introduces the issue of changing our defaults, because I don't think Pacman uninstalls optional dependencies which aren't needed anywhere else when uninstalling packages, and also installing new optional dependencies.

**Comment by karasu** [comment]
@Yoshi2889 I like your proposal. @lots0logs ?

**Comment by Warmuser** [comment]
Yo guys, I have a suggestion. What If you include both "lightdm" and "sddm" then you create a custom setting for the interested users to switch to the preferred dm. 

PS: I think Pamac is WONDERFUL. It's very useful to me.
